src/data/polynomial/ring_division.lean

```diff
@@ -665,6 +665,37 @@ lemma root_set_finite (p : T[X])
   (S : Type*) [comm_ring S] [is_domain S] [algebra T S] : (p.root_set S).finite :=
 set.to_finite _
 
+/-- The set of roots of all polynomials of bounded degree and having coefficients in a finite set
+is finite. -/
+lemma bUnion_roots_finite {R S : Type*} [semiring R] [comm_ring S] [is_domain S]
+(m : R →+* S) (d : ℕ) (U : set R) (h : U.finite) :
+  (⋃ (f : R[X]) (hf : f.nat_degree ≤ d ∧ ∀ i, (f.coeff i) ∈ U),
```

Comment:
Well, @alreadydone, here is something that I don't know how to do. Maybe you can give me a clue on how to solve that. 
Let say I want to define the function using `finset.range` instead of `fin` as you suggested (that would make the rest of the proof simpler indeed, I think), so define something like that
```lean
import ring_theory.polynomial.basic

open polynomial

variables (R : Type*) [comm_ring R] (d : ℕ) (U : set R)

noncomputable def π := λ (c : finset.range (d+1) → U),
      (finset.range (d+1)).sum (λ i, (monomial (i : ℕ)) (c i : R))
```

Then it does not compile since `i` in the sum is a `nat`. To make a `finset.range` out of it, I would need to know that `i < d+1` but `i`  does not seem to know that it comes from `finset.range (d+1)`. I could change the definition of `c` to `ℕ → U` but then it would not finite anymore...

Reply:
You could may be use some junk value instead of `c i` when `i ≥ d + 1`.

But actually I think you don't need to use `finset.sum`. The way I would do this is to apply to define a function `R[X] → finset.range (d+1) → R` and use [set.finite.of_finite_image](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.finite.of_finite_image), [set.finite.pi](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.finite.pi), and [set.finite.subset](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.finite.subset).

Reply:
> You could may be use some junk value instead of `c i` when `i ≥ d + 1`.

Yeah, I see. That's quite clever.

> But actually I think you don't need to use `finset.sum`. The way I would do this is to apply to define a function `R[X] → finset.range (d+1) → R` and use [set.finite.of_finite_image](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.finite.of_finite_image), [set.finite.pi](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.finite.pi), and [set.finite.subset](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.finite.subset).

Ok, Thanks for the suggestion. I'll see what I can get if I go that way.

Reply:
I committed a new proof along the lines that Junyan suggested. The proof is somewhat shorter and, in any case, I think easier to follow. Thanks again for the suggestion.